File: src/main/java/org/carlosalcina/java/rpg/Guerrero.java
package org.carlosalcina.java.rpg;

public class Guerrero extends Personaje {
    private final int fuerza;

    public Guerrero(String nombre, int puntosVida, int fuerza) {
        super(nombre, puntosVida);
        this.fuerza = fuerza;
    }

    public void atacar(Personaje enemigo) {
        enemigo.recibirDanio(fuerza);
        System.out.println(nombre + " ataca a " + enemigo.getNombre() + " causando " + fuerza + " puntos de daño");
    }
}

File: src/main/java/org/carlosalcina/kotlin/rpg/Guerrero.kt
package org.carlosalcina.kotlin.rpg

class Guerrero(nombre: String, puntosVida: Int, private val fuerza: Int) : Personaje(nombre, puntosVida) {
    fun atacar(enemigo: Personaje) {
        enemigo.recibirDanio(fuerza)
        println("$nombre ataca a ${enemigo.nombre} causando $fuerza puntos de daño")
    }
}